program/src/state/rules.rs

```diff
@@ -439,37 +443,38 @@ impl Rule {
             Rule::ProgramOwnedList { programs, field } => {
                 msg!("Validating ProgramOwnedList");
 
-                let key = match payload.get_pubkey(field) {
-                    Some(pubkey) => pubkey,
-                    _ => return (false, RuleSetError::MissingPayloadValue.into()),
-                };
+                let fields = field.split('|').collect::<Vec<&str>>();
 
-                let account = match accounts.get(key) {
-                    Some(account) => account,
-                    _ => return (false, RuleSetError::MissingAccount.into()),
-                };
+                for field in fields {
+                    msg!("Validating ProgramOwnedList field: {}", field);
+                    let key = match payload.get_pubkey(&field.to_string()) {
+                        Some(pubkey) => pubkey,
+                        _ => return (false, RuleSetError::MissingPayloadValue.into()),
+                    };
 
-                let data = match account.data.try_borrow() {
-                    Ok(data) => data,
-                    Err(_) => return (false, ProgramError::AccountBorrowFailed),
-                };
+                    let account = match accounts.get(key) {
+                        Some(account) => account,
+                        _ => return (false, RuleSetError::MissingAccount.into()),
+                    };
 
-                if is_zeroed(&data) {
-                    // Print helpful errors.
-                    if data.len() == 0 {
-                        msg!("Account data is empty");
-                    } else {
-                        msg!("Account data is zeroed");
-                    }
+                    let data = match account.data.try_borrow() {
+                        Ok(data) => data,
+                        Err(_) => return (false, ProgramError::AccountBorrowFailed),
+                    };
 
-                    // Account must have nonzero data to count as program-owned.
-                    (false, self.to_error())
-                } else if programs.iter().any(|program| *account.owner == *program) {
-                    // Account owner must be on the list.
-                    (true, self.to_error())
-                } else {
-                    (false, self.to_error())
+                    if is_zeroed(&data) {
+                        // Print helpful errors.
+                        if data.len() == 0 {
+                            msg!("Account data is empty");
+                        } else {
+                            msg!("Account data is zeroed");
+                        }
+                    } else if programs.iter().any(|program| *account.owner == *program) {
+                        // Account owner must be on the list.
+                        return (true, self.to_error());
+                    }
                 }
+                (false, self.to_error())
```

Comment:
maybe this default false return should possibly be moved up to be the final `else` of the `if-else` clause.

Similar to the other spot, if this found an account based on a field, but did not find an owning program on the list, it should not go back and get another field, should it?